public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -674,6 +683,41 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
       this._scrollRef?.scrollTo(0, this._prevScrollPos!);
     }
   }
+
+  getSaveModel(): Dashboard | DashboardV2Spec {
+    if (isV2Dashboard(this._initialSaveModel!)) {
+      return transformSceneToSaveModelSchemaV2(this) as DashboardV2Spec;
+    }
+    return transformSceneToSaveModel(this);
+  }
```

Comment:
This is part of the main changes, basically DashboardScene provides an interface to retrieve the save model when needed.

Reply:
would this not be cleaner with a separate class "DashboardSerializer" or something similar where we have 2 implementations (one for v1 schema and one for v2), so there would be no need for this if statement.

Reply:
yeah - it is clearer, just put it in the commit